docs/content/docs/concepts/summarization.md

```diff
@@ -0,0 +1,101 @@
+-   [What does a summary look like?](#what-does-a-summary-look-like)
+
+## Introduction
+
+This document provides a conceptual overview of summarization without going into a lot of technical or implementation details. It describes what summaries are, how / when they are generated and what do they look like. The goal is for this to be an entry point into summarization for users and developers alike.
+
+### Summary vs snapshot
+
+The terms summary and snapshot are sometimes used interchangeably. Both represent the state of a container at a point in time. They differ in some respects which are described in [this FAQ](https://fluidframework.com/docs/faq/#summarization).
+
+## Why do we need summaries?
+
+A 'summary' captures the state of a container at a point in time. Without it, a client would have to apply every operation in the op log, even if those operations no longer affected the current state (e.g. op 1 inserts 'h' and op 2 deletes 'h'). For very large op logs, this would be very expensive for clients to both process and download from the service.
+Instead, when a client opens a collaborative document, it can download a snapshot of the container, and simply process new operations from that point forward.
+
+## Who generates summaries?
+
+Summaries can be generated by any client connected in "write" mode. In the current implementation, summaries are generated by a separate non-interactive client. Using a separate client is an optimization - this client doesn't have to take local changes into account which can make the summary process more complicated. A summarizer client has the following characteristics:
+
+-   All the clients connected to the document participate in a process called "summary client election" to elect a "parent summarizer" client. Typically, it's the oldest "write" client connected to the document. The parent summarizer client spawns a "summarizer" client which is responsible for summarization.
+-   A summarizer client is like any other client connected to the document except that users cannot interact with this client, and it only works on the state it receives from other clients. It has a brand-new container with its own connection to services.
+
+> Note that if the summarizer client closes, the "summary client election" process will choose a new one, if applicable. The default "summary client election" algorithm is to select the oldest "write" client as the parent summarizer client which in turn will create the summarizer client.
+
+## When are summaries generated?
+
+The summarizer client periodically generates summary based on heuristics calculated based on configurations such as the number of user or system ops received, the amount of time a client has been idle (hasn't received any ops), the maximum time since last summary, maximum number of ops since last summary, etc. The heuristic configurations are defined by an _ISummaryConfigurationHeuristics_ interface defined in [this file](../../src/containerRuntime.ts).
```

Comment:
You can't use relative links. They either need to be absolute or you need to use "reference links" and define them differently for the in-repo file and the docs. For example:

```md
...is defined in [this file][my-name]


[my-name]: ../../src/containerRuntime.ts
```

Then in the docs file you include the middle of the source file, but omit the links, and add correct links at the bottom of the docs file. But if you can use absolute links that's easier.

Reply:
Changed them to absolute path. I am assuming you meant absolute path from the root of the repo right. I used something like "/packages/runtime/container-runtime/src/containerRuntime.ts".

Reply:
That won't be valid on the website, which is rooted under `docs`. Tyler's suggestion to use "reference links" is probably the best option here. I.e., use `[this file][my-name]` syntax in the section you're reusing, then define the following at the bottom of each respective markdown doc:

in-repo-doc
```md
[my-name]: ../../src/containerRuntime.ts
```

website doc
```md
[my-name]: <github URL for code file on main>
```

Reply:
Sounds good. Updated as per your suggestion.